bazel/grpc_build_system.bzl

```diff
@@ -298,39 +310,58 @@ def grpc_cc_test(name, srcs = [], deps = [], external_deps = [], args = [], data
         "linkstatic": linkstatic,
     }
     if uses_polling:
-        # the vanilla version of the test should run on platforms that only
-        # support a single poller
-        native.cc_test(
-            name = name,
-            testonly = True,
-            tags = (tags + [
-                "no_linux",  # linux supports multiple pollers
-            ]),
-            **args
-        )
-
-        # on linux we run the same test multiple times, once for each poller
-        for poller in POLLERS:
-            native.sh_test(
-                name = name + "@poller=" + poller,
-                data = [name] + data,
-                srcs = [
-                    "//test/core/util:run_with_poller_sh",
-                ],
-                size = size,
-                timeout = timeout,
-                args = [
-                    poller,
-                    "$(location %s)" % name,
-                ] + args["args"],
-                tags = (tags + ["no_windows", "no_mac"]),
-                exec_compatible_with = exec_compatible_with,
-                exec_properties = exec_properties,
-                shard_count = shard_count,
-                flaky = flaky,
-            )
+        # When true, EventEngines will only be exercised against one poller
+        engine_only = False
+        for engine in EVENT_ENGINES:
+            if "no_windows" not in engine["tags"] and "no_mac" not in engine["tags"]:
+                # the vanilla version of the test should run on platforms that only
+                # support a single poller
+                native.cc_test(
+                    name = name,
+                    testonly = True,
+                    tags = (tags + engine["tags"] + [
+                        "no_linux",  # linux supports multiple pollers
+                    ]),
+                    env = {"GRPC_EVENTENGINE_STRATEGY": engine["name"]},
+                    **args
+                )
+
+            if "no_linux" in engine["tags"]:
+                continue
+
+            # on linux we run the same test multiple times, once for each
+            # poller.
+            for poller in POLLERS:
+                native.sh_test(
+                    name = name + "@poller=" + poller + ",engine=" + engine["name"],
```

Comment:
is it significant that this is "engine" rather than "@engine"?

Reply:
That's a good question. I believe it's arbitrary. The tests appear to run fine. I'm happy to change it for any good reason.

Reply:
Just consistency. I have no opinion one way or the other, but the discrepancy between this and "@poller" invites questions that are probably best to avoid?

Reply:
Fair enough, I've made the change to `@poller=...@engine=...`